src/nix/util.rs

```diff
@@ -1,14 +1,18 @@
-/// Build a shell command that the user can run as-is.
-pub fn build_shell_command(program: String, args: &Vec<String>) -> String {
-    format!(
-        "{} {}",
-        program,
-        args.iter()
-            .map(|x|
-                // If the argument contains a special character, it must be
-                // quoted. We check only for "?", though.
-                if x.contains("?") { format!("\"{}\"", x) } else { x.to_string() })
-            .collect::<Vec<String>>()
-            .join(" ")
-    )
+use colored::Colorize;
+use std::io::{stderr, BufWriter, Write};
+
+/// Print a shell command that the user can run as-is.
+pub fn print_shell_command<'a>(program: &str, args: impl Iterator<Item = &'a str>) {
+    let mut stderr = BufWriter::new(stderr().lock());
+
+    let color = |s: &str| s.blue().bold();
+
+    let _ = write!(stderr, "> {}", color(program));
+
+    for arg in args {
+        let _ = write!(stderr, " {}", color(arg));
+    }
+
+    let _ = writeln!(stderr);
+    let _ = stderr.flush();
```

Comment:
In regards to the actual construction of the CLI to print, the original function body (using `format!`) seems more clear to me. What do we gain by not using `format!`?

Reply:
`format!` will allocate a string which we then copy back to stdout. By using `write!` we buffer the data just once before sending it.

But if you think this getting harder to understand I can also revert it back

Reply:
Yea, I don't think the performance gain here is worth it. Simpler code is always better.

Reply:
Sounds good, will revert this